src/jaegertracing/propagation/HeadersConfig.h

```diff
@@ -45,11 +46,16 @@ class HeadersConfig {
         const auto traceBaggageHeaderPrefix =
             utils::yaml::findOrDefault<std::string>(
                 configYAML, "traceBaggageHeaderPrefix", "");
-
-        return HeadersConfig(jaegerDebugHeader,
-                             jaegerBaggageHeader,
-                             traceContextHeaderName,
-                             traceBaggageHeaderPrefix);
+        const auto traceContextHeaderFormat =
+            utils::yaml::findOrDefault<std::string>(
+                configYAML, "TraceContextHeaderFormat", "");
+
+        return HeadersConfig(
+            jaegerDebugHeader,
+            jaegerBaggageHeader,
+            traceContextHeaderName,
+            traceBaggageHeaderPrefix,
+            traceContextHeaderFormat == "W3C" ? Format::W3C : Format::JAEGER);
```

Comment:
Q: why is this uppercase "W3C"? Where would the value come from?

Reply:
Via the configuration file, e.g.
```yaml
headers:
    TraceContextHeaderFormat: w3c
```